summoner-cli/src/Summoner/Project.hs

```diff
@@ -155,25 +157,61 @@ generateProject settingsNoUpload isOffline projectName Config{..} = do
+                        o <- queryNotNull "Nix package set GitHub owner name: "
+                        r <- queryNotNull "Nix package set GitHub repo name: "
+                        rv <- queryNotNull "Nix package set Git revision: "
+                        s  <- queryNotNull "Nix package set unpacked SHA256: "
+                        let pkgSet = NixPkgSet { npsOwner = o, npsRepo = r, npsRev = rv, npsSha = s }
+                        successMessage ("The nix package set " <> showNixPkgSet pkgSet <> " will be used in the project")
+                        pure $ Just pkgSet
+                    noDo = do
+                        successMessage ("The default nix package set " <> showNixPkgSet defaultNixPkgSet <> " will be used in the project")
+                        pure $ Just defaultNixPkgSet
+                chooseYesNo (mkDefaultYesNoPrompt "nix package set") yesDo noDo
+            Last set@(Just n) -> set <$ successMessage ("The nix package set " <> showNixPkgSet n <> " will be used in the project")
+        else pure Nothing
+
     -- get what build tool to use in the project
     -- If user chose only one during CLI, we assume to use only that one.
-    getCabalStack :: (Decision, Decision) -> IO (Bool, Bool)
-    getCabalStack = \case
-        (Idk, Idk) -> decisionToBool cCabal (mkDefaultYesNoPrompt "cabal") >>= \c ->
-            if c then decisionToBool cStack (mkDefaultYesNoPrompt "stack") >>= \s -> pure (c, s)
-            else stackMsg True >> pure (False, True)
-        (Nop, Nop) -> errorMessage "Neither cabal nor stack was chosen" >> exitFailure
-        (Yes, Yes) -> output (True, True)
-        (Yes, _)   -> output (True, False)
-        (_, Yes)   -> output (False, True)
-        (Nop, Idk) -> output (False, True)
-        (Idk, Nop) -> output (True, False)
+    getCabalStackNix :: (Decision, Decision, Decision) -> IO (Bool, Bool, Bool)
+    getCabalStackNix = \case
+        (Idk, Idk, Idk) -> do
+            c <- decisionToBool cCabal (mkDefaultYesNoPrompt "cabal")
+            s <- decisionToBool cStack (mkDefaultYesNoPrompt "stack")
+            n <- decisionToBool cNix (mkDefaultYesNoPrompt "nix")
+            when (not c && n) nixWithoutCabalError
+            output (c, s, n)
+        (Nop, Nop, Nop) -> noneOfError
+        (Yes, Yes, Yes) -> output (True, True, True)
+        (Idk, Idk, Nop) -> output (True, False, False)
+        (Idk, Idk, Yes) -> output (True, False, True)
+        (Idk, Nop, _)   -> output (True, False, False)
+        (Idk, Yes, _)   -> output (True, True, False)
+        (Nop, Idk, _)   -> output (False, True, False)
+        (Nop, Nop, Idk) -> noneOfError
+        (Nop, Nop, Yes) -> nixWithoutCabalError
+        (Nop, Yes, _)   -> output (False, True, False)
+        (Yes, Idk, _)   -> output (True, False, False)
+        (Yes, Nop, _)   -> output (True, False, False)
+        (Yes, Yes, Idk) -> output (True, True, False)
+        (Yes, Yes, Nop) -> output (True, True, False)
```

Comment:
Heh, I need to think more on this, as I'm not sure yet what is affected by Nix, sorry.